src/gitlab/configuration.ts

```diff
@@ -0,0 +1,221 @@
+  readonly workflow?: Workflow;
+  /**
+   * Groups jobs into stages. All jobs in one stage must complete before next stage is
+   * executed. Defaults to ['build', 'test', 'deploy'].
+   */
+  readonly stages?: string[];
+  readonly variables?: Record<string, any>;
+  readonly jobs?: Record<string, Job>;
+}
+
+/**
+ * CI for GitLab.
+ * A CI is a configurable automated process made up of one or more stages/jobs.
+ * @see https://docs.gitlab.com/ee/ci/yaml/
+ */
+export class CiConfiguration extends Component {
+  public readonly project: Project;
+  public readonly name: string;
+  /** Path to CI file */
+  public readonly path: string;
+  /**
+   * The workflow YAML file.
+   */
+  public readonly file: YamlFile;
+  public default?: Default;
+  /**
+   * Can be `Include` or `Include[]`. Each `Include` will be a string, or an
+   * object with properties for the method if including external YAML file. The external
+   * content will be fetched, included and evaluated along the `.gitlab-ci.yml`.
+   */
+  public readonly include: Array<Include | string> = [];
+  /**
+   * A special job used to upload static sites to Gitlab pages. Requires a `public/` directory
+   * with `artifacts.path` pointing to it.
+   */
+  public pages?: Job;
+  public readonly services: Array<Service | string> = [];
+  /**
+   * Groups jobs into stages. All jobs in one stage must complete before next stage is
+   * executed. Defaults to ['build', 'test', 'deploy'].
+   */
+  public readonly stages: string[] = [];
+  public readonly variables: Record<string, number | VariableConfig | string> = {};
+  public workflow?: Workflow;
+  public readonly jobs: Record<string, Job> = {};
+
+  constructor(project: Project, name: string, options?: CiConfigurationOptions) {
+    super(project);
+    this.project = project;
+    this.name = path.parse(name).name;
+    this.path =
+      this.name === 'gitlab-ci'
+        ? '.gitlab-ci.yml'
+        : `.gitlab/ci-templates/${name.toLocaleLowerCase()}.yml`;
+    this.file = new YamlFile(this.project, this.path, {
+      obj: () => this.renderCI(),
+    });
+    this.default = options?.default;
+    this.pages = options?.pages;
+    this.workflow = options?.workflow;
+    if (options?.stages) {
+      this.addStages(...options.stages);
+    }
+    if (options?.variables) {
+      this.addJobs(options.variables);
+    }
+    if (options?.jobs) {
+      this.addJobs(options.jobs);
+    }
+
+  }
+
+  /**
+   * Add additional yml/yaml files to the CI includes
+   * @param includes The includes to add.
+   */
+  public addIncludes(...includes: (string | Include)[]) {
+    for (const include of includes) {
+      if (this.include.includes(include)) {
+        throw new Error(`${this.name}: GitLab CI already contains ${include}.`);
+      }
+      this.include.push(include);
+    }
+  }
+
+  /**
+   * Add additional services.
+   * @param services The services to add.
+   */
+  public addServices(...services: (string | Service)[]) {
```

Comment:
Does specifying a service with a string have the same meaning as using `{ name: 'the-string' }`? if so, I feel like this is an implementation detail that probably shouldn't be covered in our API -- plus introducing type unions makes the API more obtuse when this gets compiled to other languages using JSII.

Can we simplify the API like this maybe:
```suggestion
  public addService(options: Service) {
```
(Users can call this in a for loop to add multiple services if they want)